Source/WebCore/platform/graphics/gstreamer/mse/AppendPipeline.cpp

```diff
@@ -463,6 +485,11 @@ void AppendPipeline::handleElementMessage(GstMessage* message)
         GST_DEBUG("sending drm-key-needed message from %s to the player", GST_MESSAGE_SRC_NAME(message));
         GRefPtr<GstEvent> event;
         gst_structure_get(structure, "event", GST_TYPE_EVENT, &event.outPtr(), nullptr);
+#if USE(OPENCDM)
```

Comment:
I don't understand why you require this. The logic should be able to go thru the PlayerBase as it was before, right?

Reply:
It is required to ensure the event is already fired or not. we will make the logic to avoid sending it twice

Reply:
logic updated to avoid sending of the events twice

Reply:
Why is it a problem to send the event twice? The decryptor should be able to gracefully discard it.